src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/RedesignedLanguageServerTests/CancelRequestHandlerTests.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.LanguageServerProtocol;
+using Microsoft.PowerFx.LanguageServerProtocol.Protocol;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Tests.LanguageServiceProtocol
+{
+    public partial class LanguageServerTestBase
+    {
+        [Fact]
+        public async Task TestNl2FxIsCanceledCorrectly()
+        {
+            // Arrange
+            var documentUri = "powerfx://app?context=1";
+            var engine = new Engine();
+            var symbols = new SymbolTable();
+            symbols.AddVariable("Score", FormulaType.Number);
+            var scope = engine.CreateEditorScope(symbols: symbols);
+            var scopeFactory = new TestPowerFxScopeFactory((string documentUri) => scope);
+            Init(new InitParams(scopeFactory: scopeFactory));
+            var nl2FxHandler = CreateAndConfigureNl2FxHandler();
+            nl2FxHandler.Delay = true;
+            nl2FxHandler.ThrowOnCancellation = true;
+            nl2FxHandler.Nl2FxDelayTime = 2000;
+            var payload = NL2FxMessageJson(documentUri);
+            using var source = new CancellationTokenSource();
+            source.CancelAfter(1200);

```

Comment:
>1200 [](http://example.com/codeflow?start=31&length=4)

Can we shrink this?

My concern is that we have 50k tests and counting, even 1sec/test adds up..